ocaml/parsing/parser.mly

```diff
@@ -162,25 +162,36 @@ let once_attr loc =
   mk_attr ~loc (once_ext_loc loc) (PStr [])
 
 let local_extension loc =
-  Exp.mk ~loc:Location.none
-    (Pexp_extension(local_ext_loc loc, PStr []))
+  Exp.mk (Pexp_extension(local_ext_loc loc, PStr []))
 
 let unique_extension loc =
-  Exp.mk ~loc:Location.none (Pexp_extension(unique_ext_loc loc, PStr []))
+  Exp.mk (Pexp_extension(unique_ext_loc loc, PStr []))
 
 let once_extension loc =
-  Exp.mk ~loc:Location.none (Pexp_extension(once_ext_loc loc, PStr []))
+  Exp.mk (Pexp_extension(once_ext_loc loc, PStr []))
 
-let mkexp_stack ~loc ~kwd_loc exp =
-  let loc = make_loc loc in
+let mkexp_stack ?loc ~kwd_loc exp =
+  let loc =
+    match loc with
+    | None -> Location.ghostify exp.pexp_loc
+    | Some loc -> make_loc loc
+  in
   Exp.mk ~loc (Pexp_apply(local_extension (make_loc kwd_loc), [Nolabel, exp]))
 
-let mkexp_unique ~loc ~kwd_loc exp =
-  let loc = make_loc loc in
+let mkexp_unique ?loc ~kwd_loc exp =
+  let loc =
+    match loc with
+    | None -> Location.ghostify exp.pexp_loc
+    | Some loc -> make_loc loc
+  in
   Exp.mk ~loc (Pexp_apply(unique_extension (make_loc kwd_loc), [Nolabel, exp]))
 
-let mkexp_once ~loc ~kwd_loc exp =
-  let loc = make_loc loc in
+let mkexp_once ?loc ~kwd_loc exp =
+  let loc =
+    match loc with
+    | None -> Location.ghostify exp.pexp_loc
+    | Some loc -> make_loc loc
```

Comment:
Perhaps we can move this repeated block of code from `mkexp_once/mkexp_unique/mkexp_stack` into `mkexp_with_mode`. Then, we can make the location argument to `mkexp_once/mkexp_unique/mkexp_stack` required again (which I think only affects one additional callsite &mdash; a call to `mkexp_stack` that supplies the location anyway).

Reply:
Please see my other comments before addressing this comment. It may no longer be needed.

Reply:
Yes, I believe this is resolved because no longer needed.